pkg/bucket/replication/replication.go

```diff
@@ -199,7 +210,7 @@ func (c Config) HasActiveRules(prefix string, recursive bool) bool {
 				continue
 			}
 			// If recursive, we can skip this rule if it doesn't match the tested prefix.
-			if recursive && !strings.HasPrefix(rule.Filter.Prefix, prefix) {
+			if recursive && !strings.HasPrefix(rule.Filter.Prefix, prefix) && !strings.HasPrefix(prefix, rule.Filter.Prefix) {
```

Comment:
is this fixing a bug? @poornas - can we add a rationale for this check?

Reply:
@poornas you need to use rule.GetPrefix() here